dts/bindings/video/video-interfaces.yaml

```
@@ -0,0 +1,93 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for video interfaces.
+
+include: base.yaml
+
+properties:
+  bus-type:
+    type: int
+    enum:
+      - 1 # Parallel
+      - 2 # MIPI CSI-2 D-PHY
+      - 3 # MIPI CSI-2 C-PHY
+      - 4 # MIPI CSI-1
+    description:
+      Data bus type.
+
+  bus-width:
+    type: int
+    maximum: 64
+    description:
+      Number of data lines actively used, valid for the parallel busses.
+
+  data-shift:
+    type: int
+    maximum: 64
+    description:
+      On the parallel data busses, if bus-width is used to specify the number of
+      data lines, data-shift can be used to specify which data lines are used,
+      e.g. "bus-width=<8>; data-shift=<2>;" means, that lines 9:2 are used.
+
+  hsync-active:
+    type: int
+    enum:
+      - 0
+      - 1
+    description:
+      Active state of the HSYNC signal, 0/1 for LOW/HIGH respectively.
+
+  vsync-active:
+    type: int
+    enum:
+      - 0
+      - 1
+    description:
+      Active state of the VSYNC signal, 0/1 for LOW/HIGH respectively. Note,
+      that if HSYNC and VSYNC polarities are not specified, embedded
+      synchronization may be required, where supported.
+
+  pclk-sample:
+    type: int
+    enum:
+      - 0
+      - 1
+      - 2
```

Same as above, some opportunity to give the enum names if desired:
```suggestion
      - "falling"
      - "rising"
      - "both"
```

With a devicetree API like this to refer to them:

```C
DT_ENUM_IDX(node, falling) // 0
DT_ENUM_IDX(node, rising)  // 1
DT_ENUM_IDX(node, both)    // 2
```